### DISCUSSION TITLE
Rendering of source text diff in target string doesn’t show the real changes in source code

### DISCUSSION THREAD

**Discussion by Geeyun-JY3** [WeblateOrg/weblate#7168] [Q&A] [answered]
It only shows the difference between the changed string itself and the most similar string among the TM.
(Looks like that. Is it related to automatic translation? Enable `fuzzy` - Add as needing edit as automatic translation mode? [Weblate project](https://hosted.weblate.org/projects/weblate/) enable [Weblate itself as the source of machine translations](https://docs.weblate.org/en/latest/admin/machine.html#weblate)? I know nothing about this. Just a speculation)
For example, [Source string description case-insensitive search.](https://hosted.weblate.org/translate/weblate/documentation/zh_Hans/?checksum=15be9d556465fc39#history)
- In [GitHub](https://github.com/WeblateOrg/weblate/commit/9ac63dd864e3444dbab886359ab5ba4f65136583#diff-3eb290919f61bddd29299720477d0001574593aa098ddaf1e41011adb8ebfc3aL36-R36):
  
  ~~Comment~~ string case-insensitive search.
  <ins>Source</ins> string <ins>description</ins> case-insensitive search.

  ```diff
  - Comment string case-insensitive search.
  + Source string description case-insensitive search.
  ```

- But in Weblate (translate page of target string):
  
  Source string case-insensitive search.
  Source string <ins>description</ins> case-insensitive search.

  ```diff
  - Source string case-insensitive search.
  + Source string description case-insensitive search.
  ```

The `String age` and `Source string age` attribute in `String information` panel of that string are also updated. It seems that it is considered as a new string by Weblate.

And is it related to specific file format? For now, I find that only strings in `.po` file would be like this.
For example, strings in `.json` file: [this string](https://hosted.weblate.org/translate/weblate/hosted-error-pages/zh_Hans/?checksum=92048f2895471655) in [Hosted: Error pages](https://hosted.weblate.org/projects/weblate/hosted-error-pages/) component.

Is it intended?

**Comment by nijel** [comment, marked as answer]
This is how strings are handled by Gettext - strings are identified by the source (and optionally context). When the string changes, the old one stays (it becomes obsolete) and a new one is added as fuzzy.

**Reply by Geeyun-JY3** [reply]
Got it. So the comments and suggestions in Weblate of the string would be lost if the source of string is changed, right?

**Reply by nijel** [reply]
Yes. Actually, what also often happens as well is that both strings stay - one location is changed to a different context and old string still stays somewhere else.

**Reply by Geeyun-JY3** [reply]
Why I find that the old source (text) is removed instead of staying in source code when the strings changes?

And another question, when the string changes, does the new source text come from TM of the project?

**Reply by nijel** [reply]
All this is done by Gettext, Weblate is not involved at all. Check documentation for xgettext and msgmerge for more details.

**Reply by Geeyun-JY3** [reply]
OK. I will look it up.
